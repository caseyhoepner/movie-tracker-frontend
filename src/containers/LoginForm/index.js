import React, { Component } from 'react'

export class LoginForm extends Component {
  constructor() {
    super()
    this.state = {

    }
  }

  render() {
    return(
      <form>
        <input />
      </form>
    )
  }
}

export default LoginForm;