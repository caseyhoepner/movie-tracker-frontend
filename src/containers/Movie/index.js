import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';

import * as API from '../../utils'
import { toggleFavorite } from '../../actions';
import './Movie.css';

// make functional
export class Movie extends Component {
  handleClick = async () => {
    const { movie_id, overview, poster_path, release_date, title, vote_average, isLoggedIn, toggleFavorite } = this.props;
    const movie = {
      user_id: isLoggedIn,
      movie_id: movie_id,
      overview,
      poster_path,
      release_date,
      title,
      vote_average,
    }

    await API.addFavorite(movie)
    toggleFavorite(movie_id)
  }

  render() {
    let favoriteBtn;
    const { poster_path, movie_id, isLoggedIn, isFavorite } = this.props;

    if (isLoggedIn) {
      favoriteBtn =
        <button className='m-fav-container'
          onClick={this.handleClick}
        >
          <p className='m-fav-text'>FAVORITE</p>
          <div
            className={`m-fav-btn ${isFavorite ? 'red' : ''}`}>
          </div>
        </button>
    } else {
      favoriteBtn =
        <NavLink className='m-fav-container'
          to='/login'>
          <p className='m-fav-text'>FAVORITE</p>
          <div className='m-fav-btn'>
          </div>
        </NavLink>
    }

    return (
      <div className='m-movie'>
        <Link to={`/movie/${movie_id}`}>
          <div className='m-movie-poster'>
            <img
              className='m-poster'
              src={`http://image.tmdb.org/t/p/w342//${poster_path}`}
              alt='Click here for movie details.'/>
          </div>
        </Link>
        { favoriteBtn }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  isLoggedIn: state.user.id,
})

export const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId)),
})

Movie.propTypes = {
  movie_id: PropTypes.number.isRequired,
  overview: PropTypes.string.isRequired,
  poster_path: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  vote_average: PropTypes.number,
  isFavorite: PropTypes.bool.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Movie);