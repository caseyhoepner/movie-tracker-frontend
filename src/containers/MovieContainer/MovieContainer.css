.movie-container {
  height: 100%;
  width: 80%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-flow: wrap;
  margin: 40px auto;
}

/*.movie-container::after {
  content: "";
  flex: auto;
}*/