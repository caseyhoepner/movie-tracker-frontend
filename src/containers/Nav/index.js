import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeUser, filterMovies } from '../../actions';
import userIcon from '../../assets/images/margot.png';
import './Nav.css';

export const Nav = (props) => {
  const { userLoggedIn, removeUser, movies, filterMovies } = props

  const handleHover = (event) => {
    if(!event.target.classList.contains('active')) {
      event.target.classList.add('active')
    } else {
      event.target.classList.remove('active')
    }
  }

  return (
    <nav className="nav-container">
      <NavLink
        onClick={() => filterMovies(movies)}
        to="/favorites"
        className="fav-btn-group"
      >
        <div
          className="fav-btn-nav"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <span className="num-favs">0</span>
          <p
            className={`favorites-text`}
          >
            FAVORITES
          </p>
        </div>
      </NavLink>
      <NavLink
        to="/login"
        className={userLoggedIn ? "hidden" : "show"}
      >
        <div
          className="login-btn"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <img
            className="login-icon"
            src={ userIcon }
            alt='Click here to login'/>
          <p className="login-text">
            LOGIN
          </p>
        </div>
      </NavLink>
      <NavLink
        className={userLoggedIn ? "logout-btn" : "hidden"}
        onClick={removeUser}
        to='/'
      >
        <img
          className="login-icon"
          src={ userIcon }
          alt='Click here to login'/>
        <p className="login-text">
          LOGOUT
        </p>
      </NavLink>
    </nav>
  )
}

export const mapStateToProps = (state) => ({
  userLoggedIn: state.user.id,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  removeUser: () => dispatch(removeUser()),
  filterMovies: (movies) => dispatch(filterMovies(movies))
})

Nav.propTypes = {
  removeUser: PropTypes.func.isRequired,
  userLoggedIn: PropTypes.number
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav);