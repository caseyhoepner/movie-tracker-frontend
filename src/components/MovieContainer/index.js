import React from 'react';
import PropTypes from 'prop-types';
import './MovieContainer.css';
import Movie from '../Movie';
import { connect } from 'react-redux';


const MovieContainer = (props) => {
  const movies = props.movies.map(movie => {
    return <Movie { ...movie }/>
  })

  return (
    <div className='movie-container'>
      { movies }
    </div>
  )
}

const mapStateToProps = (state) => ({
  movies: state.movies
})


export default connect(mapStateToProps)(MovieContainer);