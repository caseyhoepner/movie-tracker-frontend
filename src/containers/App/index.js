import React, { Component } from 'react';
import { Link, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as API from '../../utils';
import apiKey from '../../API-key.js';
import { addMovies, toggleFavorite } from '../../actions';
import MovieContainer from '../MovieContainer';
import Nav from '../Nav';
import LoginForm from '../LoginForm'
import MovieDetails from '../../components/MovieDetails';
import wesIcon from '../../assets/images/wes.png';
import './App.css';

export class App extends Component {
  async componentDidMount() {
    const movies = await API.fetchMovies(apiKey)
    await this.props.addMovies(movies)
  }

  render() {
    const { user } = this.props
    return (
      <div className="a-App">
        <header className="a-header-container">
          <Link to='/' className="a-logo">
            <img
              src={ wesIcon }
              className="a-logo-icon"
              alt="An icon of Wes Anderson's face"
            />
            <h1 className='a-header-title'>WesTracker</h1>
          </Link>
          <Nav />
        </header>
        <Route
          exact path='/'
          component={ MovieContainer }
        />
        <Route
          exact path='/login'
          component={ LoginForm }
        />
        <Route path='/movie/:id' render={({match}) => {
          const { id } = match.params;
          const movie = this.props.movies.find(movie => (
            movie.id === parseInt(id, 10)))
          return (
            <MovieDetails {...movie} />
          )}} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies)),
  toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId))
})

App.propTypes = {
  user: PropTypes.object.isRequired,
  addMovies: PropTypes.func.isRequired
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));