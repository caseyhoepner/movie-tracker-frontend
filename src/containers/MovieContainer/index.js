import React from 'react';
import PropTypes from 'prop-types';
import './MovieContainer.css';
import Movie from '../../components/Movie';
import { connect } from 'react-redux';


export const MovieContainer = (props) => {
  const movies = props.movies.map(movie => {
    return <Movie { ...movie } key={movie.id} />
  })

  return (
    <div className='movie-container'>
      { movies }
    </div>
  )
}

const mapStateToProps = (state) => ({
  movies: state.movies
})

MovieContainer.propTypes = {
  movies: PropTypes.array.isRequired
}

export default connect(mapStateToProps)(MovieContainer);