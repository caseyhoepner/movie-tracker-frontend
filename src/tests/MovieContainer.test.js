import React from 'react';
import { shallow } from 'enzyme';
import { MovieContainer, mapStateToProps } from '../containers/MovieContainer';
import { mockMovies } from './testMocks';

describe('MovieContainer', () => {
  describe('MovieContainer Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<MovieContainer movies={ mockMovies }/>);
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should parse the movies from state', () => {
      const mockState = {
        movies: mockMovies
      }
      const expected = mockMovies
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps.movies).toEqual(expected)
    })
  })
})