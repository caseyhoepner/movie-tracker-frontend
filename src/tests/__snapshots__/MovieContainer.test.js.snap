// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer MovieContainer Component should match the snapshot 1`] = `
<div
  className="movie-container"
>
  <Movie
    backdrop_path="backdrop-path1.jpg"
    id={1}
    isFavorite={false}
    key="1"
    overview="Great film."
    poster_path="poster-path1.jpg"
    release_date="2014-02-26"
    title="The Grand Budapest Hotel"
  />
  <Movie
    backdrop_path="backdrop-path2.jpg"
    id={2}
    isFavorite={false}
    key="2"
    overview="Real good film."
    poster_path="poster-path2.jpg"
    release_date="2015-03-27"
    title="The Life Aquatic with Steve Zissou"
  />
</div>
`;
